src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -99,3 +99,10 @@ def is_valid_connection_string(connection_string):
             return False
         return True
     return False
+
+
+def get_store_name_from_connection_string(connection_string):
+    if is_valid_connection_string(connection_string):
+        segments = connection_string.split(';')
+        return segments[0].split('//')[1].split('.')[0]
```

Comment:
There is work item about the connection string validation in CLI. As of now, we expect the connection string to exactly like "Endpoint=abv.azconfig.io;Id=xxx;Secret=xxx". The order of endpoint, id and secret is fixed, which doesn't need to. So for this helper method, we can ignore the order, split by ";" and convert the three segments to dictionary. 

```
segments= dict(seg.split("=", 1) for seg in connection.split(";"))
endpoint = segments.get("Endpoint")
if not endpoint:
  return None
store = endpoint.split("//")[1].split('.')[0]
```

Reply:
done

---
In reply to: [364503307](https://github.com/Azure/azure-cli/pull/11773#discussion_r364503307) [](ancestors = 364503307)